scripts/tags_and_languages.py

```diff
@@ -0,0 +1,338 @@
+            str_split(unnest(hf_dataset.{tag_type}_tags), ':')[2] AS {tag_type}_tag
+        FROM 'hf://datasets/openfoodfacts/product-database/food.parquet' hf_dataset
+        WHERE {tag_type}_tags IS NOT NULL
+    ),
+    products_unknown_tags AS (
+        SELECT
+            products_tags.code,
+            products_tags.lang,
+            products_tags.{tag_type},
+            products_tags.{tag_type}_lc_and_tag,
+            products_tags.{tag_type}_tag,
+            taxonomy_tags.tag,
+            taxonomy_tags.lc,
+            taxonomy_tags.lc_tag_name,
+            taxonomy_tags.lc_tag_name_concat
+        FROM products_tags
+        LEFT JOIN {tag_type}_tags taxonomy_tags
+        ON products_tags.{tag_type}_lc_and_tag = taxonomy_tags.tag
+        WHERE taxonomy_tags.lc_tag_name IS NULL
+    )
+    SELECT
+        products_unknown_tags.code AS products_code,
+        products_unknown_tags.lang AS products_lang,
+        products_unknown_tags.{tag_type} AS products_{tag_type},
+        products_unknown_tags.{tag_type}_lc_and_tag AS products_{tag_type}_unknown_tag,
+        products_unknown_tags.{tag_type}_tag AS products_{tag_type}_unknown_tag_name,
+        taxonomy_tags.tag AS taxonomy_tag_id,
+        taxonomy_tags.lc AS taxonomy_tag_lc,
+        taxonomy_tags.lc_tag_name AS taxonomy_tag_name,
+        taxonomy_tags.lc_tag_name_concat AS taxonomy_tag
+    FROM products_unknown_tags
+    LEFT JOIN {tag_type}_tags taxonomy_tags
+    ON products_unknown_tags.{tag_type}_tag = taxonomy_tags.lc_tag_name
+    WHERE taxonomy_tags.lc_tag_name IS NOT NULL
+    AND taxonomy_tags.lc != 'xx'
+    AND products_unknown_tags.{tag_type}_lc_and_tag != taxonomy_tags.lc_tag_name_concat
+    ''').df()
+    # remark:
+    #   AND products_unknown_tags.{tag_type}_lc_and_tag != taxonomy_tags.lc_tag_name_concat -> to prevent fr:angleterre and fr:angleterre
+
+    print(f"prepare log table - time {datetime.now()-time_start}")
+    conn_db = duckdb.connect(database='tags_and_languages.db', read_only=False)
+    conn_db.execute(f'''
+    DROP TABLE IF EXISTS products_to_update_{tag_type}
+    ''')
+    conn_db.execute(f'''
+    CREATE TABLE products_to_update_{tag_type} (
+        code TEXT,
+        lang TEXT,
+        old_tags_text TEXT,
+        new_tags_text TEXT,
+        updated_tags_text BOOLEAN,
+        updated BOOLEAN
+    )
+    ''')
+    for _, row in tags_to_update_df.iterrows():
+        code = row[0]
+        lang = row[1]
+        
+        existing_row = conn_db.execute(f"SELECT * FROM products_to_update_{tag_type} WHERE code = '{code}'").fetchone()
+        
+        if existing_row:
+            # Update the existing row
+            # existing_row[2] is tags as text already updated
+            # row[3] is unknown tag with lc
+            # row[4] is unknown tag
+            # row[8] is row[4] tag found in another language
+            new_tags_text, updated_tags_text = update_tags(existing_row[3], row[3], row[4], row[8])
```

Comment:
Using named parameters helps readability

```suggestion
            # row[3] is unknown tag with lc
            # row[4] is unknown tag
            # row[8] is row[4] tag found in another language
            new_tags_text, updated_tags_text = update_tags(tags_text=existing_row[3], old_tag_with_lc=row[3], old_tag=row[4], new_tag=row[8])
```